doc/how_to/custom_components/reactive_esm/reactive_esm_layout.md

```diff
@@ -0,0 +1,382 @@
+
+:::::{tab-set}
+
+::::{tab-item} `Viewer`
+
+```{pyodide}
+import panel as pn
+from panel.viewable import Viewer, Layoutable
+from panel.custom import Children
+from panel.layout.base import NamedListLike
+
+pn.extension()
+
+
+class ListLikeLayout(NamedListLike, Layoutable, Viewer):
+    objects = Children()
+
+    def __init__(self, *args, **params):
+        super().__init__(*args, **params)
+
+        layoutable_params = {name: self.param[name] for name in Layoutable.param}
+        self._layout = pn.Column(
+            **layoutable_params,
+        )
+        self._objects()
+
+    def __panel__(self):
+        return self._layout
+
+    @pn.depends("objects", watch=True)
+    def _objects(self):
+        objects = []
+        for object in self.objects:
+            objects.append(object)
+            objects.append(
+                pn.pane.HTML(
+                    styles={"width": "calc(100% - 15px)", "border-top": "3px dotted #bbb"},
+                    height=10,
+                )
+            )
+
+        self._layout[:] = objects
+
+
+ListLikeLayout(
+    "I love beat boxing",
+    "https://upload.wikimedia.org/wikipedia/commons/d/d3/Beatboxset1_pepouni.ogg",
+    "Yes I do!",
+    styles={"border": "2px solid lightgray"},
+).servable()
+```
+
+You must list `NamedListLike, Layoutable, Viewer` in exactly that order when you define the class! Other combinations might not work.
+
+::::
+
+::::{tab-item} `JSComponent`
+
+```{pyodide}
+import panel as pn
+import param
+from panel.custom import JSComponent
+from panel.layout.base import NamedListLike
+
+pn.extension()
+
+
+class ListLikeLayout(NamedListLike, JSComponent):
+    objects = param.List()
+
+    _esm = """
+    export function render({ model }) {
+      const div = document.createElement('div')
+      let objects = model.get_child("objects")
+
+      objects.forEach((object, index) => {
+        div.appendChild(object);
+
+        // If it's not the last object, add a divider
+        if (index < objects.length - 1) {
+            const divider = document.createElement("div");
+            divider.className = "divider";
+            div.appendChild(divider);
+        }
+        });
+      return div
+    }"""
+
+    _stylesheets = [
+        """
+.divider {border-top: 3px dotted #bbb};
+"""
+    ]
+
+
+ListLikeLayout(
+    "I love beat boxing",
+    "https://upload.wikimedia.org/wikipedia/commons/d/d3/Beatboxset1_pepouni.ogg",
+    "Yes I do!",
+    styles={"border": "2px solid lightgray"},
+).servable()
+```
+
+You must list `NamedListLike, JSComponent` in exactly that order when you define the class! The other
+way around `JSComponent, NamedListLike` will not work.
+
+::::
+
+::::{tab-item} `ReactComponent`
+
+```{pyodide}
+import panel as pn
+
+from panel.custom import Children, ReactComponent
+
+class Example(ReactComponent):
+
+    objects = Children()
+
+    _esm = """
+    export function render({ model }) {
```

Comment:
This does not work. I simple cannot figure out how to loop over the `objects`. Seems like its assumed you want to insert them together?

Reply:
I can try to have another go at this but it was quite difficult to correctly update each one in place. That said for this type of templated setup this does seem important.

Reply:
I believe you dont normally want to insert them together. If you do, then just use a Column.

Reply:
Done.